includes/class-wc-gateway-komoju-ipn-handler.php

```diff
@@ -4,76 +4,93 @@
 }
 
 include_once( 'class-wc-gateway-komoju-response.php' );
+include_once( 'class-wc-gateway-komoju-webhook-event.php');
 
 /**
  * Handles responses from Komoju IPN
  */
 class WC_Gateway_Komoju_IPN_Handler extends WC_Gateway_Komoju_Response {
 
-	protected $notify_url;
+	protected $webhookSecretToken;
 	protected $secret_key;
 	protected $invoice_prefix;
 	/**
 	 * Constructor
 	 */
-	public function __construct( $sandbox = false, $notify_url = '', $secret_key = '', $invoice_prefix = ''  ) {
+	public function __construct( $webhookSecretToken = '', $secret_key = '', $invoice_prefix = ''  ) {
 		add_action( 'woocommerce_api_wc_gateway_komoju', array( $this, 'check_response' ) );
 		add_action( 'valid-komoju-standard-ipn-request', array( $this, 'valid_response' ) );
 
-		$this->sandbox        	= $sandbox;
-		$this->notify_url	  	= $notify_url;
-		$this->secret_key	  	= $secret_key;
-		$this->invoice_prefix	= $invoice_prefix;
+		$this->webhookSecretToken	  	= $webhookSecretToken;
+		$this->secret_key	   	        = $secret_key;
+		$this->invoice_prefix			= $invoice_prefix;
 	}
 
 	/**
 	 * Check for Komoju IPN Response
 	 */
 	public function check_response() {
-		if ( ! empty( $_GET ) && $this->validate_hmac() ) {
-			$posted = wp_unslash( $_GET );
+		$entityBody = file_get_contents('php://input');
 
-			do_action( "valid-komoju-standard-ipn-request", $posted['transaction'] );
+		if ( ! empty( $entityBody ) && $this->validate_hmac($entityBody) ) {
+			$webhookEvent = new WC_Gateway_Komoju_Webhook_Event($entityBody);

```

Comment:
encapsulating the webhook request into it's own class, so if future changes are needed to support the API we can (hopefully) just make them in the `WC_Gateway_Komoju_Webhook_Event` class